api/obs/api/routes/tiles.py

```diff
@@ -38,11 +41,27 @@ async def tiles(req, zoom: int, x: int, y: str):
         tile = get_tile(req.app.config.TILES_FILE, int(zoom), int(x), int(y))
 
     else:
+        user_id = None
+        username = req.ctx.get_single_arg("user", default=None)
+        if username is not None:
+            if req.ctx.user is None or req.ctx.user.username != username:
+                raise Forbidden()
```

Comment:
Tested this before even looking at the code :) Was not disappointed to get a "forbidden".